client/src/components/MultiJurisdictionAudit/AASetup/Contests/ContestForm.tsx

```diff
@@ -0,0 +1,304 @@
+          name: '',
+          numVotes: '',
+        },
+        {
+          id: '',
+          name: '',
+          numVotes: '',
+        },
+      ],
+    },
+  ]
+  const { electionId } = useParams<{ electionId: string }>()
+  const [contests, updateContests] = useContests(electionId)
+  const jurisdictions = useJurisdictions(electionId)
+
+  if (!contests) return null // Still loading
+  const filteredContests = contests.filter(c => c.isTargeted === isTargeted)
+
+  const isBatch = auditType === 'BATCH_COMPARISON'
+  // const isBallotComparison = auditType === 'BALLOT_COMPARISON'
+
+  /* istanbul ignore next */
+  if (isBatch && !isTargeted && nextStage.activate) nextStage.activate() // skip to next stage if on opportunistic contests screen and during a batch audit (until batch audits support multiple contests)
+
+  const submit = async (values: { contests: IContest[] }) => {
+    const response = await updateContests(values.contests)
+    // TEST TODO
+    /* istanbul ignore next */
+    if (!response) return
+    /* istanbul ignore else */
+    if (nextStage.activate) nextStage.activate()
+    else throw new Error('Wrong menuItems passed in: activate() is missing')
+  }
+  return (
+    <Formik
+      initialValues={{
+        contests: filteredContests.length ? filteredContests : contestValues,
+      }}
+      validationSchema={schema}
+      enableReinitialize
+      onSubmit={submit}
+    >
+      {({
+        values,
+        handleSubmit,
+        setFieldValue,
+      }: FormikProps<{ contests: IContest[] }>) => (
+        <form data-testid="form-one">
+          <FormWrapper
+            title={isTargeted ? 'Target Contests' : 'Opportunistic Contests'}
+          >
+            <FieldArray
+              name="contests"
+              render={contestsArrayHelpers => (
+                <>
+                  {values.contests.map((contest: IContest, i: number) => {
+                    const jurisdictionOptions = jurisdictions.map(j => ({
+                      title: j.name,
+                      value: j.id,
+                      checked: contest.jurisdictionIds.indexOf(j.id) > -1,
+                    }))
+                    return (
+                      /* eslint-disable react/no-array-index-key */
+                      <Card key={i}>
+                        <FormSection
+                          label={`Contest ${
+                            values.contests.length > 1 ? i + 1 : ''
+                          } Info`}
+                          description="Enter the name of the contest that will drive the audit."
+                        >
+                          <br />
+                          <label htmlFor={`contests[${i}].name`}>
+                            Contest {values.contests.length > 1 ? i + 1 : ''}{' '}
+                            Name
+                            <Field
+                              id={`contests[${i}].name`}
+                              name={`contests[${i}].name`}
```

Comment:
@benadida No, this is how Formik handles objects and arrays of form data. Every field has a unique name attached to it which is a string constructed from its location in the state, like this.

Reply:
I'm still confused, cause you've got `values.contests.map((contest, ..)` above, which makes me think `contests[${i}]` is the same as `contest`.. but that's not the case? If so, that's confusing, maybe different variable names would help?

Reply:
You are correct that the value of `contest` and `contests[i]` is the same. The string ``contests[${i}].name`` though is what is required for Formik to know which form element goes with which piece of its internal state. The strings passed to `id` and `name` are basically used as arguments for a getter function inside of Formik. If you passed in the string `'contest.name'` it would look on the root state object for a key called `contest` and things would get very confused, haha.

Reply:
oh! Duh. These are strings. My bad. I get it now.